src/pharmpy/modeling/odes.py

```diff
@@ -765,6 +845,8 @@ def set_transit_compartments(model: Model, n: int, keep_depot: bool = True):
             n -= 1
             cb.add_flow(new_comp, comp, rate)
             comp = new_comp
+        comp = cb.set_bioavailability(comp, dosing_comp.bioavailability)
+        dosing_comp = cb.set_bioavailability(dosing_comp, sympy.Integer(1))
```

Comment:
Will this be a problem if you e.g. first have one transit and then add another (i.e. call `set_transit_compartments` multiple times)? Then `dosing_comp` will no longer have the bioavailability?

Reply:
When calling `set_transit_compartments` multiple times the dose compartment is changed for each call and hence the bioavailability is moved for each call. I might be misunderstanding your question but through my own testing this did not cause any issues.
Please let me know if you wish to take a closer look at this.

Reply:
Yes you're right, misread where in the overall code this was, my mistake!